decode.go

```diff
@@ -71,6 +77,9 @@ type UnmarshalOptions struct {
 	// DiscardUnknown specifies whether to discard unknown fields instead of
 	// returning an error.
 	DiscardUnknown bool
+
+	// CelEnv is the CEL environment to use for evaluating CEL expressions.
```

Comment:
Would need more documentation: why is this here, what is it for, why is it part of a yaml library